### PR TITLE
Throw if accessing a closed realm

### PR DESCRIPTION
**Pull request by kristiandupont**

### CODE REVIEW

File: src/shared_realm.hpp [hunk 1]
```diff
@@ -386,6 +387,11 @@ class IncorrectThreadException : public std::logic_error {
     IncorrectThreadException() : std::logic_error("Realm accessed from incorrect thread.") {}
 };
 
+class ClosedRealmException : public std::runtime_error {
```
Comment (tgoyne) [inline src/shared_realm.hpp]: This is a logic error.

**Comment by bdash** [comment]
This could do with tests.

**Comment by alazier** [comment]
We should either make these changes on master, or base this pr on the js branch.

**Comment by bdash** [comment]
Changes that aren't specific to a given binding should really be merged to master first. We're trying to eliminate differences, not introduce more.

**Pull request merged**